### PR TITLE
Introduce more context into test callbacks

### LINKED ISSUE #9812: Extend QuarkusTestCallbacks with a context like JUnit Jupiter
**Description**
Currently you only get the test instance, but it would be nice to have more context, like TestClass, TestMethod.

This request is result of Zulip talk with  @geoand
https://quarkusio.zulipchat.com/#narrow/stream/187030-users/topic/QuarkusTestExtension

**Implementation ideas**
```
public interface QuarkusTestBeforeEachCallback {

    void beforeEach(QuarkusCallbackContext context);
}

public class QuarkusCallbackContext {

    private Object testInstance;

    private Class<?> testClass;
    
    private Method testMethod;
    ...
}
```

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #9812

### CODE REVIEW

File: test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java [hunk 1]
```diff
@@ -292,6 +269,45 @@ public void beforeEach(ExtensionContext context) throws Exception {
...
+            return;
+        }
+        if (!failedBoot) {
+            popMockContext();
+            for (Object afterEachCallback : afterEachCallbacks) {
+                Map.Entry<Class<?>, ?> tuple = createQuarkusTestMethodContextTuple(context);
+                afterEachCallback.getClass().getMethod("afterEach", tuple.getKey())
+                        .invoke(afterEachCallback, tuple.getValue());
+            }
+            boolean nativeImageTest = isNativeTest(context);
```
Comment (aloubyansky) [inline test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:284]: Isn't this going to be always false?

Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java:284]: That must be leftover - I just moved the method. I'll take a look tomorrow

Comment (geoand) [reply test-framework/junit5/src/main/java/io/quarkus/test/junit/QuarkusTestExtension.java]: Good catch, removed

**Comment by geoand** [comment]
@stuartwdouglas are you OK with this one?

**Review by stuartwdouglas — APPROVED** [review]

**Pull request merged**